app/products/calls/hooks.ts

```diff
@@ -155,3 +163,66 @@ export const useCallsAdjustment = (serverUrl: string, channelId: string): number
         (joinCallBannerVisible ? JOIN_CALL_BAR_HEIGHT + 8 : 0) +
         callsIncomingAdjustment;
 };
+
+export const useHostControlsAvailable = () => {
+    const [isAdmin, setIsAdmin] = useState(false);
+
+    const currentCall = getCurrentCall();
+    const serverUrl = currentCall?.serverUrl || '';
+    const config = getCallsConfig(serverUrl);
+    const allowed = isHostControlsAllowed(config);
+
+    useEffect(() => {
+        const getUser = async () => {
+            const database = DatabaseManager.serverDatabases[serverUrl]?.database;
+            if (!database) {
+                return;
+            }
+            const user = await getCurrentUser(database);
+            setIsAdmin(isSystemAdmin(user?.roles || ''));
```

Comment:
so this is a one time thing, if the roles of the user changed while the call is ongoing it won't  update, perhaps you want to observe the user ? up to you

Reply:
If a user is changed to admin while the call screen (or participant list) is open, then they won't suddenly get host controls, correct. They would need to reopen the call screen (or participant list).
But imo that's an acceptable tradeoff for simplicity, given how rare that would happen + cause a UX issue.